### PR TITLE
improve(cross-chain-oracle): Allow publishPrice to be called for any DVM resolved price

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

`OracleBase.publishPrice` can currently only be called after a price is requested, which usually comes via a cross-chain price request. This is unnecessarily restrictive as the `OracleSpokes` should be able to make all prices available that have been resolved on the DVM, even if they weren't originally requested via the `OracleSpoke` on the child network.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/core/contracts/cross-chain-oracle/OracleBase.sol [hunk 1]
```diff
@@ -71,11 +73,9 @@ abstract contract OracleBase {
...
         bytes32 priceRequestId = _encodePriceRequest(identifier, time, ancillaryData);
         Price storage lookup = prices[priceRequestId];
-        if (lookup.state == RequestState.Requested) {
-            lookup.price = price;
-            lookup.state = RequestState.Resolved;
-            emit PushedPrice(identifier, time, ancillaryData, lookup.price, priceRequestId);
-        }
+        lookup.price = price;
+        lookup.state = RequestState.Resolved;
+        emit PushedPrice(identifier, time, ancillaryData, lookup.price, priceRequestId);
```
Comment (chrismaree) [inline packages/core/contracts/cross-chain-oracle/OracleBase.sol:78]: I think there needs to be a bit more logic here:
1) if the `OracleBase` already has a price it should not change the stored price. 
2) we should only emit an event on the first time the function is called.

you can simply add the following to the top of this function:
```
if (lookup.state == RequestState.Resolved) return;
```

this way you exit early, don't override the price and don't double emit

**Review by chrismaree — APPROVED** [review]
This looks great to me! I left one comment about early exit to ensure we: a) don't override the previous price and b) only emit once.

File: packages/core/contracts/cross-chain-oracle/OracleBase.sol (hunk 1 above)
Comment (mrice32) [reply packages/core/contracts/cross-chain-oracle/OracleBase.sol:78]: Agree with this comment.

**Review by mrice32 — APPROVED** [review]
+1 on this PR since this allows us to resolve prices faster than the L2 -> L1 bridges might allow as long as we're willing to lose the final fee.

LGTM once @chrismaree's comment is resolved.

**Pull request merged**